### PR TITLE
V6

### PR DESCRIPTION
**Pull request by erlendvollset**
- Remove support for legacy auth
- Remove deprecated properties and parameters

### CODE REVIEW

**Review by qTipTip — APPROVED** [review]
LGTMlops 🚀

File: cognite/client/_cognite_client.py
Comment (haakonvt) [inline cognite/client/_cognite_client.py:1]: Remove:
```
@property  # TODO (v6.0.0): Delete this whole property
    def extraction_pipeline_runs(self) -> ExtractionPipelineRunsAPI:
```

Comment (haakonvt) [inline cognite/client/_cognite_client.py:1]: Remove:
```
@property  # TODO (v6.0.0): Delete this whole property
    def datapoints(self) -> DatapointsAPI:
        if int(self.version.split(".")[0]) >= 6:
```

File: CHANGELOG.md
Comment (haakonvt) [inline CHANGELOG.md:1]: Remove deprecated camelCase parameter support from:
```
cognite/client/_api/extractionpipelines.py (one use)

cognite/client/data_classes/shared.py (two uses)
```

**Review by haakonvt — CHANGES_REQUESTED** [review]
There are a few properties & parameters that should be removed once moving to major version 6. I think it would make sense to hold off on the major version bump for as long as possible, and rather add warnings or similar?

We need to keep in mind each major version bump will keep a loooot of our customers from receiving updates to the SDK for a long time since we do not keep updating v5 with minor/patch releases after moving to 6 😬

**Review by haakonvt — CHANGES_REQUESTED** [review]
Let's also add an entry in `MIGRATION_GUIDE.md` 🚀

**Review by haakonvt — APPROVED** [review]
LGTM. Could you fix the bad date on line 35 in the change log? It should not be November 😅 
```
## [5.11.0] - 17-11-23
```

**Pull request merged**